internal/core/components/rewriter.go

```diff
@@ -0,0 +1,91 @@
+package components
+
+import (
+	"context"
+	"runtime"
+	"sync"
+
+	"github.com/api7/apisix-seed/internal/core/entity"
+	"github.com/api7/apisix-seed/internal/core/storer"
+	"github.com/api7/apisix-seed/internal/discoverer"
+	"github.com/api7/apisix-seed/internal/utils"
+)
+
+type Rewriter struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	// Limit the number of simultaneously update
+	sem chan struct{}
+}
+
+func (r *Rewriter) Init() {
+	r.ctx, r.cancel = context.WithCancel(context.TODO())
+	r.sem = make(chan struct{}, runtime.GOMAXPROCS(0)+10)
```

Comment:
Why use a magic number 10?

Reply:
Here I refer to the settings of some other projects. In fact, I'm not sure about the number of concurrency control...

Reply:
We can add a comment for it, including the link to that project